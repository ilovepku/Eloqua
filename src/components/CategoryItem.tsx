import React from 'react'
import {View, TouchableOpacity, Image, Text} from 'react-native'
import {useNavigation} from '@react-navigation/native'
import tailwind from 'tailwind-rn'

import {CategoriesQueryCategory} from '../types/graphql'
import {ASSETS_URL} from '../settings'

interface Props {
  category: CategoriesQueryCategory
}

const CategoryItem: React.FC<Props> = ({
  category: {name, id, icon_filename: iconFilename},
}) => {
  const navigation = useNavigation()

  const navigateToFilteredPiecesListScreen = (): void => {
    navigation.navigate('FilteredPiecesListScreen', {
      title: name,
      categoryIdFilter: `category-${id}`,
    })
  }

  return (
    <TouchableOpacity
      style={tailwind('w-1/3 h-24 mt-4 items-center')}
      onPress={navigateToFilteredPiecesListScreen}
    >
      <Image
        source={{
          uri: `${ASSETS_URL}/icons%2F${iconFilename}?alt=media`,
        }}
        style={tailwind('h-20 w-20 rounded-full mb-1')}
      />
      <View style={tailwind('flex-1 text-gray-700')}>
        <Text style={tailwind('text-center text-xs')}>{name}</Text>
      </View>
    </TouchableOpacity>
  )
}

export default CategoryItem
