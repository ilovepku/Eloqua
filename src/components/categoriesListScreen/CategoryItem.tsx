import React from 'react'
import {View, TouchableOpacity, Image, Text} from 'react-native'
import {useNavigation} from '@react-navigation/native'
import tailwind from 'tailwind-rn'

import {CategoriesQuery_category} from '../../types/graphql'
import {ASSETS_URL} from '../../settings'

interface Props {
  category: CategoriesQuery_category
}

export default function CategoryItem({
  category: {name, id, icon_filename},
}: Props) {
  const navigation = useNavigation()

  const navigateToFilteredPiecesListScreen = () => {
    navigation.navigate('FilteredPiecesListScreen', {
      title: name,
      category_id_filter: `category-${id}`,
    })
  }

  return (
    <TouchableOpacity
      style={tailwind('w-1/3 h-24 mt-4 items-center')}
      onPress={navigateToFilteredPiecesListScreen}
    >
      <Image
        source={{
          uri: `${ASSETS_URL}/icons%2F${icon_filename}?alt=media`,
        }}
        style={tailwind('h-20 w-20 rounded-full mb-1')}
      />
      <View style={tailwind('flex-1 text-gray-700')}>
        <Text style={tailwind('text-center text-xs')}>{name}</Text>
      </View>
    </TouchableOpacity>
  )
}
