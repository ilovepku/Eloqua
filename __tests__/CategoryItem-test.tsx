import 'react-native';
import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import CategoryItem from '../src/components/categoriesListScreen/CategoryItem';
import {useNavigation} from '@react-navigation/native';

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: jest.fn(),
  };
});

beforeEach(() => {
  // @ts-ignore
  useNavigation.mockReset();
});

it('renders correctly', () => {
  const tree = render(
    <CategoryItem
      category={{
        __typename: 'categories',
        name: 'Acknowledgement',
        id: 1,
        icon_filename: 'acknowledgement.png',
      }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('onPress: navigates to FilteredPiecesListScreen with correct params', async () => {
  const mockNavigate = jest.fn();
  // @ts-ignore
  useNavigation.mockImplementation(() => ({navigate: mockNavigate}));

  const {getByText} = render(
    <CategoryItem
      category={{
        __typename: 'categories',
        name: 'Acknowledgement',
        id: 1,
        icon_filename: 'acknowledgement.png',
      }}
    />,
  );

  const clickable = getByText(/Acknowledgement/i);
  fireEvent.press(clickable);

  await waitFor(() => expect(mockNavigate).toHaveBeenCalledTimes(1));
  expect(mockNavigate).toHaveBeenCalledWith('FilteredPiecesListScreen', {
    category_id_filter: 'category-1',
    title: 'Acknowledgement',
  });
});
