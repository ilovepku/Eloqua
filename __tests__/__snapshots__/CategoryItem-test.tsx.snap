// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 96,
      "marginTop": 16,
      "opacity": 1,
      "width": "33.333333%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://firebasestorage.googleapis.com/v0/b/speech-pwa.appspot.com/o/icons%2Facknowledgement.png?alt=media",
      }
    }
    style={
      Object {
        "borderBottomLeftRadius": 9999,
        "borderBottomRightRadius": 9999,
        "borderTopLeftRadius": 9999,
        "borderTopRightRadius": 9999,
        "height": 80,
        "marginBottom": 4,
        "width": 80,
      }
    }
  />
  <View
    style={
      Object {
        "color": "rgba(74, 85, 104, 1)",
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    >
      Acknowledgement
    </Text>
  </View>
</View>
`;
